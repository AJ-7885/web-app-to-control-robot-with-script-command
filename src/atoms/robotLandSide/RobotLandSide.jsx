import React from 'react';
import robot from '../../assets/robot.png';
import { RobotLand, RobotArea, Robot } from './robotLandSide.style';

export const RobotLandSide = ({robotPosition}) => {
  return (
    <RobotLand>
      <RobotArea>
        <Robot position={robotPosition} src={robot} alt="robot" />
      </RobotArea>
    </RobotLand>
  );
};
