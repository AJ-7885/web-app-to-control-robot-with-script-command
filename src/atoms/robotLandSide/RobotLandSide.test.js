import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme/build';
import { RobotLandSide } from './RobotLandSide';

const mockRobotPosition = { top: 0, left: 0, direction: 2 };

test('renders component and check the content', () => {
  const component = mount(<RobotLandSide robotPosition={mockRobotPosition} />);
  expect(component.find('div').length).toBe(2);
});

test('renders correctly', () => {
  const component = renderer
    .create(<RobotLandSide robotPosition={mockRobotPosition} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
