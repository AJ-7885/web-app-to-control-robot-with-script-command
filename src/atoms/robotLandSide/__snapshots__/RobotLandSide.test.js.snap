// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm htywmC"
>
  <div
    className="sc-AxirZ fgsdab"
  >
    <img
      alt="robot"
      className="sc-AxiKw tJrRc"
      src="robot.png"
    />
  </div>
</div>
`;
